src/main/java/com/mraof/minestuck/event/ServerEventHandler.java

```diff
@@ -232,6 +240,32 @@ public static void onPlayerInjured(LivingHurtEvent event)
 		}
 	}
 	
+	@SubscribeEvent(priority=EventPriority.LOW)
+	public static void onBlockBreak(BlockEvent.BreakEvent event)
+	{
+		Player player = event.getPlayer();
+		BlockPos pos = event.getPos();
+		Level level = player.level;
+		ItemStack stack = player.getMainHandItem(); //pickaxes cannot be swung in the offhand
+		BlockState state = event.getState();
+		
+		if(stack.getItem() == MSItems.MINE_AND_GRIST.get() && !level.isClientSide && stack.isCorrectToolForDrops(state)) //will only drop grist instead of an item if it can be mined by the pick
+		{
+			ItemStack droppedStack = state.getCloneItemStack(null, level, pos, player);
```

Comment:
I wonder if we could gristify the actual block loot drops instead of the pick-block result.

Reply:
Resolved by you!